TEST_APPS/testcases/test_spi_com/README.md

```diff
@@ -0,0 +1,71 @@
+SPI Communication Test
+=============
+
+Testcases
+---------
+
+### test_spi_com
+
+**Description**  
+This is the SPI communication test which verifies various SPI configuration variants. This test is based on Ice Tea framework. Information how to install and use Ice Tea can be found [here](https://github.com/ARMmbed/icetea).  
+This test transfers data between SPI master and SPI slave.
+Two boards are required with SPI peripheral available and slave/master mode support.    
+
+To run the test using same boards as master and slave perform the following steps:  
+ 1. Adjust configuration to your needs in `./TEST_APPS/device/spi_common/mbed_app.json.`  
+ 2. Connect SPI interfaces as configured in `./TEST_APPS/device/spi_common/mbed_app.json`.  
+ 3. Execute the following command:  
+ `mbed test -m BOARD -t TOOLCHAIN --icetea --app-config ./TEST_APPS/device/spi_common/mbed_app.json -n SPI_COM*`
+
+Note:  
+Above command will run all SPI communication test cases. To run only one test case type its full name.
+
+Note:  
+It is possible to use two different boards for SPI master and SPI slave, but the test must be  run using IceTea (not mbed test). Check the manual for more details.  
+
+Note:  
+Remember to connect also ground pins between boards.  
+You may consider to add a pull-up resistor to the clock line. While working on NUCLEO-F429ZI driver some clock noises have been found in half-duplex mode and while testing different clock modes.  
+
+**Mbed app config**  
+ - `SPI_MASTER/SLAVE_SS`: master/slave slave select pin,    
+ - `SPI_MASTER/SLAVE_MOSI`: master/slave mosi pin,  
+ - `SPI_MASTER/SLAVE_MISO`: master/slave miso pin,  
+ - `SPI_MASTER/SLAVE_MISO`: master/slave clk pin,  
+ - `SPI_MASTER_HALF_DUPLEX_DATA`: master/slave data line in half-duplex mode,  
+ - `SPI_MASTER_SS_ACTIVE_HIGH`: `true` if SS is asserted when set to logic 1, `false` otherwise,  
+ - `SPI_MASTER/SLAVE_DEBUG`: master/slave debug mode (0 or 1). If debug is enabled then tested configuration is dumped and transmission buffers are dumped in case of failure.  
+ - `SPI_MASTER/SLAVE_DELAY`: Delay before the transmission in ms. We need to ensure that slave is ready for transmission.  
+ - `SPI_MASTER/SLAVE_TRANSMISSION_START_BTN`: it is possible to define button to trigger the transmission manually. In such case set `SPI_MASTER_DELAY` to 0 and define `SPI_MASTER/SLAVE_TRANSMISSION_START_BTN`. If not used then set to `NC`.  
+ - `SPI_MASTER/SLAVE_TRANSMISSION_START_LED`: set led which indicates delay before the transmission. Set `NC` if not used. 
+
+**Which SPI features are verified by the test?**  
```

Comment:
which test case cover which area needs to be better explained.
Maybe this section could be moved to after the test scenario diagram. and be put into a table?
you can use[ netsocket tests readme](https://github.com/ARMmbed/mbed-os/tree/master/TESTS/netsocket) as an example.
If `Preconditions` `test steps` and `Expected results` of each test case need explanation, those can be added

Reply:
Fixed as suggested.